Cargo.toml

```diff
@@ -108,6 +108,7 @@ criterion = { version = "0.4", features = [ "async_tokio" ] }
 crossbeam = "0.8"
 crucible-agent-client = { git = "https://github.com/oxidecomputer/crucible", rev = "0e2bea7741c6bdc34e8e08addb941ee31e4ac9ec" }
 crucible-client-types = { git = "https://github.com/oxidecomputer/crucible", rev = "0e2bea7741c6bdc34e8e08addb941ee31e4ac9ec" }
+crucible-smf = { git = "https://github.com/oxidecomputer/crucible", rev = "5f06c7ebc320c19846670e9d745a4d44f78901b4" }
```

Comment:
This `rev` is different from the one used in `crucible-agent-client` and `crucible-client-types`. This seems to be okay, although it means two checkouts of the same repo. I tried bumping the other two to match and ran into multiple type definition problems that looked like propolis needed an update first, maybe? I'm happy to address this in whatever way makes sense:

a) leave as-is, and crucible will be updated the next time it needs to be in the normal course of it being updated
b) leave as-is on this PR, but try to follow up and update crucible ASAP
c) update crucible as a part of this PR, even if that means changes not directly related to MGS

Reply:
My vote is for (b), fwiw